File: array_and_hashing/cpp/groupAnagrams.cpp
#include <vector>
#include <string>
#include <map>
using namespace std;

vector<vector<string>> groupAnagrams(vector<string>& strs) {
    map<vector<int>, vector<string>> groups;
    vector<vector<string>> result;
    
    for(auto word: strs){
        vector<int> count(26, 0);
        for(auto c : word){
            count[int(c) - int('a')]++;
        }
        groups[count].push_back(word);
    }

    for(auto list: groups){
        result.push_back(list.second);
    }
    return result;
}

File: array_and_hashing/swift/groupAnagrams.swift
func groupAnagrams(_ strs: [String]) -> [[String]] {
    var res = [[Int]:[String]]()
    let baseChar : Character = "a"
    for word in strs{
        var count : [Int] = Array(repeating: 0, count: 26)
        for ch in word{
            let index = Int(ch.asciiValue! - baseChar.asciiValue!)
            count[index] += 1
        }
        res[count, default: []].append(word)
    }
    return Array(res.values)
}